libraries/AP_Math/AP_Math_Float.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * This file is the current standard math implementation until the "DBL_MATH" build flag is set.
+ * With this template implementation all double promotions are avoided, 
+ * which can happen because of lazy integer usage in ArduPilot.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include "AP_Math_Common.h"
+
+
+/*
+ * Should be in the std library, but isn't for many platforms
+ */
+#undef isfinite
+namespace std {
+    template<class T>
+    bool isfinite(const T& val) {
+        static_assert(std::is_floating_point<T>::value, "ERROR - isfinite(): template parameter not of type float\n");
+        return (isnan(val) || isinf(val)) ? false : true;
+    }
+};
```

Comment:
humn.. we should had already fixed this... just make sure to include cmath. Or is there anything wrong with it still?